include/nccl_ofi_idpool.h

```diff
@@ -5,90 +5,77 @@
 #ifndef NCCL_OFI_IDPOOL_H_
 #define NCCL_OFI_IDPOOL_H_
 
-#include <pthread.h>
-#include <stdint.h>
+#include <algorithm>
+#include <mutex>
+#include <vector>
+
 
 /*
  * Pool of IDs, used to keep track of communicator IDs and MR keys.
  */
-typedef struct nccl_ofi_idpool {
-	/* Size of the id pool (number of IDs) */
-	size_t size;
+class nccl_ofi_idpool_t {
+public:
+	/* Disable default constructor to require passing in a size */
+	nccl_ofi_idpool_t() = delete;
```

Comment:
Since you provide the constructor below, you shouldn't need to explicitly delete the default constructor.  I think.

On the other hand, we probably do want to delete the implicit copy constructor and just not have one.  I'm not sure I can come up with reasonable copy semantics for an id pool without way overcomplicating the id pool.

Reply:
I tested it and you are correct, I wouldn't need to explicitly delete the default constructor. And makes sense for the implicit copy constructor, I'll delete it along with the copy assignment operator:
```
nccl_ofi_idpool_t(const nccl_ofi_idpool_t&) = delete;
nccl_ofi_idpool_t& operator=(const nccl_ofi_idpool_t&) = delete;
```

Reply:
Changed in force push.